formats/json/utils.go

```diff
@@ -0,0 +1,176 @@
+package json
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/santhosh-tekuri/jsonschema/v5"
+)
+
+func exploreSchema(jsonSchema *jsonschema.Schema) map[string]*jsonschema.Schema {
+	baseLocation := jsonSchema.Location
+	exploredSchemas := make(map[string]*jsonschema.Schema, 10)
+	explore(jsonSchema, exploredSchemas, baseLocation)
```

Comment:
mutation of  ?

Reply:
exploredSchemas

Reply:
exploredSchemas acts as the visited Set during DFS, so I don't think we can avoid mutation of the map, since it's purpose is to mark visited nodes to avoid infinite loops(via cyclic references) during the DFS.